src/components/GlobalPreferences/Notifications/NotificationsLogin.js

```diff
@@ -0,0 +1,178 @@
+import React, { useState, useCallback } from 'react'
+import PropTypes from 'prop-types'
+import {
+  Box,
+  Button,
+  GU,
+  TextInput,
+  Info,
+  IconMail,
+  IconCross,
+  IconCheck,
+  textStyle,
+  useTheme,
+} from '@aragon/ui'
+import { login } from './notification-service-api'
+import { network } from '../../../environment'
+import { validateEmail } from '../../../utils'
+import notificationSvg from './notifications.svg'
+import throttle from 'lodash.throttle'
+
+export default function NotificationsLogin({ dao, authState, onEmailChange }) {
+  const [inputEmail, setInputEmail] = useState('')
+  const [emailInvalid, setEmailInvalid] = useState(null)
+  const [apiError, setApiError] = useState(null)
+  // The notifications API expects mainnet or rinkeby. This deviates from web3's getNetworkType which returns main
+  const ethNetwork = network.type === 'main' ? 'mainnet' : 'rinkeby'
+
+  const handleEmailValidation = useCallback(
+    throttle(email => {
+      if (email.length > 3 && email.includes('@')) {
+        setEmailInvalid(!validateEmail(email))
+      }
+    }, 600),
+    [setEmailInvalid]
+  )
+  const handleEmailBlur = useCallback(
+    e => {
+      const email = e.target.value
+      setEmailInvalid(!validateEmail(email))
+    },
+    [setEmailInvalid]
+  )
+
+  const handleEmailChange = useCallback(
+    e => {
+      const email = e.target.value
+      setInputEmail(email)
+      handleEmailValidation(email)
+    },
+    [handleEmailValidation, setInputEmail]
+  )
+
+  const handleLogin = async e => {
+    e && e.preventDefault()
+    try {
+      await login({ email: inputEmail, dao, network: ethNetwork })
+      onEmailChange(inputEmail)
+    } catch (e) {
+      setApiError(e.message)
+      console.error('Failed to login', e)
+    }
+  }
+
+  const theme = useTheme()
+
+  return (
+    <Box heading="Email notifications">
+      <NotificationImage />
+      <form onSubmit={handleLogin}>
+        <div
+          css={`
+            display: grid;
+            align-items: end;
+            grid-template-columns: 1fr auto;
+            grid-gap: ${GU}px ${2 * GU}px;
+            margin-bottom: ${2 * GU}px;
+          `}
+        >
+          <label
+            css={`
+              color: ${theme.surfaceContentSecondary};
+              display: block;
+            `}
+          >
+            Email address
+            <TextInput
+              css={`
+                border-color: ${emailInvalid === true
+                  ? theme.error
+                  : theme.border};
+              `}
+              adornment={
```

Comment:
Just a reminder the input still loses focus after the first input (it is very confusing and annoying) cc @bpierre

Reply:
Resolved with https://github.com/aragon/aragon/pull/904/commits/987d05b2f2063ef4b0e5f5603062c44c40db45d9